### PR TITLE
[TP 147318] - fix crash that occurs in the test app

### PR DESCRIPTION
**Pull request by ndurell**
Adds some guards in track and identify to prevent test app crash.

# Testing
I tested this manually with the test by sending a push after stopping the app.
When starting the app from a dead state it would crash on launch due a missing API key.

I also added some unit test to increase coverage.

### CODE REVIEW

File: Sources/KlaviyoSwift/Klaviyo.swift [hunk 1]
```diff
@@ -228,15 +228,15 @@ public class Klaviyo : NSObject {
         let customerPropertiesDict = updatePropertiesDictionary(propDictionary: customerProperties)
         assertPropertyTypes(properties: propertiesDict)
         
+        guard let apiKey = apiKey else {
+            environment.logger.error("Track event called before API key was set.")
```
Comment (kennyklaviyo) [inline Sources/KlaviyoSwift/Klaviyo.swift:232]: should we replace existing usages of `print` with this logger too?

Comment (ndurell) [reply Sources/KlaviyoSwift/Klaviyo.swift:232]: Eventually yah...trying to keep this focused though. I can file a ticket for it.

**Review by kennyklaviyo — APPROVED** [review]

File: klaviyo-swift-sdk.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -22,6 +22,7 @@
 
 /* Begin PBXBuildFile section */
 		7C375BC028E77149008605B3 /* KlaviyoTestUtils.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7C375BBF28E77149008605B3 /* KlaviyoTestUtils.swift */; };
+		7C54A46F2902F01E00611E4E /* LoggerClient.swift in Sources */ = {isa = PBXBuildFile; fileRef = 7C54A46E2902F01E00611E4E /* LoggerClient.swift */; };
```
Comment (jordan-griffin) [inline klaviyo-swift-sdk.xcodeproj/project.pbxproj:25]: Question: Should we start gitignoring this file now that we are using SPM? Locals should be able to regenerate it with `swift package generate-xcodeproj` right?

Comment (jordan-griffin) [reply klaviyo-swift-sdk.xcodeproj/project.pbxproj:25]: I'm not entirely sure how safe that is though...

**Review by jordan-griffin — APPROVED** [review]

**Pull request merged**